internal/schema/schema.go

```diff
@@ -382,34 +392,77 @@ func GetSchema(ctx context.Context, db queries.DBTX, opts ...GetSchemaOpt) (Sche
 		opt(&options)
 	}
 
+	nameFilter, err := buildNameFilter(options)
+	if err != nil {
+		return Schema{}, fmt.Errorf("building name filter: %w", err)
+	}
+
 	return (&schemaFetcher{
 		q:                      queries.New(db),
 		goroutineRunnerFactory: goroutineRunnerFactory,
-		nameFilter:             buildNameFilter(options),
+		nameFilter:             nameFilter,
 	}).getSchema(ctx)
 }
 
-func buildNameFilter(options getSchemaOptions) nameFilter {
-	if len(options.includeSchemas) == 0 {
+func buildNameFilter(options getSchemaOptions) (nameFilter, error) {
+	if intersection := intersect(options.includeSchemas, options.excludeSchemas); len(intersection) > 0 {
+		return nil, fmt.Errorf("schemas %v are both included and excluded", intersection)
+	}
+
+	includeFilter := buildIncludeSchemasFilter(options.includeSchemas)
+	excludeFilter := buildExcludeSchemasFilter(options.excludeSchemas)
+	return andNameFilter(includeFilter, excludeFilter), nil
```

Comment:
We technically don't need to build a `excludeSchemasFilter` if there is an `includeSchemasFilter`, however I think the flow of control here is a bit simpler.

Everything will also still work if we got rid of the intersection check with this method!